### PR TITLE
Chinese Translation - capabilities&methodology Chapter #1067

### PR DESCRIPTION
**Pull request by chengxicn**
Chinese:capabilities&methodology 2020

### CODE REVIEW

**Comment by chengxicn** [comment]
Tried submiting the chapters but failed on "Test website".  Not sure how to solve it.

File: src/content/zh-CN/2020/capabilities.md [hunk 1]
```diff
@@ -0,0 +1,332 @@
+---
+part_number: II
+chapter_number: 13
+title: Capabilities
```
Comment (rviscomi) [inline src/content/zh-CN/2020/capabilities.md:4]: Could you also translate all of this metadata?

**Review by rviscomi — COMMENTED** [review]
Thanks for working on this @chengxicn!

File: src/templates/zh-CN/2020/methodology.html [hunk 2]
```diff
@@ -0,0 +1,475 @@
+{% extends "base/2019/methodology.html" %}
```
Comment (bazzadp) [inline src/templates/zh-CN/2020/methodology.html:1]: This has now moved to reflect the fact it's no longer just for 2019 (This is what is causing your test failure btw):

```suggestion
{% extends "base/methodology.html" %}
```

Comment (chengxicn) [reply src/templates/zh-CN/2020/methodology.html]: Wow,thank you very much!

File: src/content/zh-CN/2020/capabilities.md (hunk 1 above)
Comment (chengxicn) [reply src/content/zh-CN/2020/capabilities.md]: translated

File: src/content/zh-CN/2020/capabilities.md [hunk 3]
```diff
@@ -0,0 +1,332 @@
...
+
+归功于[存储管理API](https://developer.mozilla.org/en-US/docs/Web/API/StorageManager)，它是[WHATWG 存储动态标准](https://storage.spec.whatwg.org/#storagemanager)的一部分，使得浏览器在这方面的表现不再像一个黑盒子了。这个API允许开发者估计剩余的可用空间，并选择加入[持久性存储](https://web.dev/persistent-storage/)，这意味着当磁盘空间不足时，浏览器不会清除网站的数据。因此，该API在`navigator` 对象上引入了一个新的`StorageManager`接口，目前在Chrome、Edge和Firefox上都支持。
+
+### 预估可用存储
+
+开发者可以通过调用`navigator.storage.repair()`来估计可用的存储空间。该方法返回一个承诺(promise)，解析为一个具有两个属性的对象：`usage`显示应用程序使用的字节数，`quota`包含可用的最大字节数。
+
+{{ figure_markup(
+  image="storage_manager_api_estimate.png",
+  caption="使用存储管理 API的预估方法的页数。",
```
Comment (bazzadp) [inline src/content/zh-CN/2020/capabilities.md:81]: This changed recently in #1623 

```suggestion
  alt="使用存储管理 API的预估方法的页数。",
  caption="使用存储管理 API的预估方法的页数。",
```

File: src/content/zh-CN/2020/capabilities.md [hunk 4]
```diff
@@ -0,0 +1,332 @@
...
+
+Chrome浏览器从2016年开始支持存储管理API，Firefox以及基于Chromium的新Edge浏览器是从2017年开始支持的。HTTP Archive数据显示，27,056个桌面页面（0.49%）和34,042个移动页面（0.48%）使用了该API。在2020年之间，存储管理器API的使用量不断增长。这也使得该接口成为本章中最常用的API。
+
+### 选择加入持久存储
+
+Web存储有两类。"尽力"和 "持久"，第一种是默认的。当设备的存储空间不足时，浏览器会自动尝试释放出尽力存储空间。例如，基于Firefox和Chromium的浏览器会从最近使用最少的源回收存储。然而，这带来了丢失关键数据的风险。为了防止回收，开发者可以选择持久存储。在这种情况下，即使空间不足浏览器也不会清除存储。用户仍然可以选择手动清理存储。要选择持久存储，开发者需要调用`navigator.storage.persist()`方法。根据浏览器和网站的参与情况，可能会出现权限提示，或者自动接受和拒绝请求。
+
+{{ figure_markup(
+  image="storage_manager_api_persist.png",
+  caption="使用存储管理 API的持久方法的页面数。",
```
Comment (bazzadp) [inline src/content/zh-CN/2020/capabilities.md:97]:
```suggestion
  alt="使用存储管理 API的持久方法的页面数。",
  caption="使用存储管理 API的持久方法的页面数。",
```

File: src/content/zh-CN/2020/capabilities.md [hunk 5]
```diff
@@ -0,0 +1,332 @@
...
+从Chrome 80到83版本，Fugu团队首次在初期试用中尝试使用通知触发器，之后由于开发者反馈不足而暂停开发。从2020年10月发布的Chrome 86开始，API再次进入初期试用阶段。这也解释了通知触发API的使用数据，在2020年3月左右的第一次初期试用中的调用达到了峰值，在Chrome浏览器中占0.000032%的页面加载量。
+
+## 屏幕唤醒锁定API
+
+为了节约能源，移动设备会将屏幕背光变暗，最终关闭设备的显示屏，这在大多数情况下是合理的。然而，在某些情况下，用户可能希望应用程序明确地保持显示屏的唤醒，例如，在做饭时阅读菜谱或观看演示时。[屏幕唤醒锁定API](https://developer.mozilla.org/en-US/docs/Web/API/Screen_Wake_Lock_API)([W3C工作草案](https://www.w3.org/TR/screen-wake-lock/))通过提供一种保持屏幕开启的机制来解决这个问题。
+
+`navigator.wakeLock.request()`方法创建一个唤醒保持锁。这个方法需要一个`WakeLockType`参数。在未来，唤醒保持API可以提供其他的保持类型，比如关闭屏幕但保持CPU开启。目前，该API只支持屏幕唤醒的保持，所以只有一个可选的参数，默认值为`screen`。该方法返回一个承诺，该承诺解析为一个`WakeLockSentinel`对象。开发者需要存储这个引用，以便以后调用它的`release()`方法，释放屏幕唤醒保持锁。当标签页处于非活动状态，或者用户将窗口最小化时，浏览器会自动释放保持锁。另外，浏览器可能会拒绝请求，拒绝承诺，例如由于电池电量不足的时候。
+
+
+
```
Comment (bazzadp) [inline src/content/zh-CN/2020/capabilities.md:157]:
```suggestion
```

File: src/content/zh-CN/2020/capabilities.md [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+{{ figure_markup(
+  image="screen_wake_lock_api.png",
+  caption="使用屏幕唤醒锁定 API的页面数量。",
```
Comment (bazzadp) [inline src/content/zh-CN/2020/capabilities.md:160]:
```suggestion
  alt="使用屏幕唤醒锁定 API的页面数量。",
  caption="使用屏幕唤醒锁定 API的页面数量。",
```

File: src/content/zh-CN/2020/capabilities.md [hunk 7]
```diff
@@ -0,0 +1,332 @@
...
+registration.periodicSync.register('articles', {
+  minInterval: 24 * 60 * 60 * 1000 // one day
+});
+```
+
+### 定期同步间隔的响应
+
+对于间隔的每一个周期，如果设备在线，浏览器会触发服务工作程序的 "periodicsync "事件。然后，服务工作程序脚本可以执行必要的步骤来同步数据。
+
+```
```
Comment (bazzadp) [inline src/content/zh-CN/2020/capabilities.md:206]:
```suggestion
```js
```

File: src/content/zh-CN/2020/capabilities.md [hunk 8]
```diff
@@ -0,0 +1,332 @@
...
+  }
+});
+```
+
+在撰写本文时，只有基于Chromium的浏览器实现了这个API。在这些浏览器上，必须先安装应用程序（即添加到主屏幕上），然后才能使用该API。网站的[网站参与度评分](https://www.chromium.org/developers/design-documents/site-engagement)定义了是否可以调用周期性同步事件以及调用的频率。在目前保守的实现中，网站每天可以同步一次内容。
+
+
+{{ figure_markup(
+  image="periodic_background_sync_api.png",
+  caption="使用周期性后台同步 API的页面数量。",
```
Comment (bazzadp) [inline src/content/zh-CN/2020/capabilities.md:219]:
```suggestion
  alt="使用周期性后台同步 API的页面数量。",
  caption="使用周期性后台同步 API的页面数量。",
```

File: src/content/zh-CN/2020/capabilities.md [hunk 9]
```diff
@@ -0,0 +1,332 @@
...
+self.addEventListener('periodicsync', (event) => {
+  if (event.tag === 'articles') {
+    event.waitUntil(syncStuff());
+  }
+});
+```
+
+在撰写本文时，只有基于Chromium的浏览器实现了这个API。在这些浏览器上，必须先安装应用程序（即添加到主屏幕上），然后才能使用该API。网站的[网站参与度评分](https://www.chromium.org/developers/design-documents/site-engagement)定义了是否可以调用周期性同步事件以及调用的频率。在目前保守的实现中，网站每天可以同步一次内容。
+
+
```
Comment (bazzadp) [inline src/content/zh-CN/2020/capabilities.md:216]:
```suggestion

```

File: src/content/zh-CN/2020/capabilities.md [hunk 10]
```diff
@@ -0,0 +1,332 @@
...
+目前，该接口的使用率非常低。在2020年，HTTP Archive监测到的网页中只有一两个页面使用了这个API。
+
+## 同原生应用商店集成
+
+PWA是一种多功能的应用模式。然而，在某些情况下，提供一个单独的原生应用可能仍然是有意义的：例如，如果应用需要使用网络上无法使用，或者基于应用开发团队的编程经验的功能。当用户已经安装了一个原生应用时，应用可能不希望发送两次通知或推广安装相应的PWA。
+
+为了检测用户在系统中是否已经有相关的原生应用或PWA，开发者可以在`navigator`对象上使用[getInstalledRelatedApps()方法](https://web.dev/get-installed-related-apps/)([WICG社区组报告草案](https://wicg.github.io/get-installed-related-apps/spec/))。该方法目前由基于Chromium的浏览器提供，适用于Android和通用Windows平台（UWP）应用。开发者需要调整本地应用捆绑以引用网站，并将本地应用的信息添加到PWA的Web应用清单中。调用`getInstalledRelatedApps()`方法将返回用户设备上安装的应用列表。
+
+```
```
Comment (bazzadp) [inline src/content/zh-CN/2020/capabilities.md:235]:
```suggestion
```js
```

File: src/content/zh-CN/2020/capabilities.md [hunk 11]
```diff
@@ -0,0 +1,332 @@
...
+  caption='使用 WebSocketStreams 的 Chrome 浏览器中页面加载的百分比。<br>(来源: <a href="https://chromestatus.com/metrics/feature/timeline/popularity/3018">WebSocketStream</a>)',
+  description="WebSocketStreams 使用情况图，基于使用该功能的 Chrome 浏览器中页面加载的百分比。它显示了2020年6月和7月的峰值，其中API在大约0.0008%的页面加载中被使用。",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTxqot9ALgxcgOVJntkzIKnkpo3idIPy-tL0t_nzC5BwFuq0ThgK5OXOYVVOpama4vB2EyggX813d33/pubchart?oid=1714443590&format=interactive",
+  sheets_gid="691106754"
+  )
+}}
+
+WebSocketStream API已经完成了第一次初期试用，现在又回到了实验阶段。这也解释了为什么目前这个API的使用率很低，几乎无法衡量。
+
+
```
Comment (bazzadp) [inline src/content/zh-CN/2020/capabilities.md:301]:
```suggestion

```

File: src/templates/zh-CN/2020/methodology.html [hunk 12]
```diff
@@ -0,0 +1,475 @@
...
+  <span class="code-function">APPROX_QUANTILES</span>(<span class="code-function">ROUND</span>(bytesJs <span class="code-keyword">/</span> <span class="code-number">1024</span>, <span class="code-number">2</span>), <span class="code-number">1000</span>)[<span class="code-function">OFFSET</span>(percentile <span class="code-keyword">*</span> <span class="code-number">10</span>)] <span class="code-keyword">AS</span> js_kbytes
+<span class="code-keyword">FROM</span>
+  <span class="code-string">`httparchive.summary_pages.2019_07_01_*`</span>,
+  <span class="code-function">UNNEST</span>([<span class="code-number">10</span>, <span class="code-number">25</span>, <span class="code-number">50</span>, <span class="code-number">75</span>, <span class="code-number">90</span>]) <span class="code-keyword">AS</span> percentile
+<span class="code-keyword">GROUP BY</span>
+  percentile,
+  client
+<span class="code-keyword">ORDER BY</span>
+  percentile,
+  client</code></pre>
```
Comment (bazzadp) [inline src/templates/zh-CN/2020/methodology.html:105]: Changed in #1594 

```suggestion
          {# To generate this markup temporarily add a ```sql code block to a chapter and generate that chapter and you'll get the HTML #}
          <pre><code class="sql language-sql"><span class="comment">#standardSQL</span>
<span class="comment"># 01_01b: Distribution of JS bytes by client</span>
<span class="keyword">SELECT</span>
  percentile,
  _TABLE_SUFFIX <span class="keyword">AS</span> client,
  <span class="function call">APPROX_QUANTILES</span>(<span class="function call">ROUND</span>(bytesJs / <span class="constant numeric">1024</span>, <span class="constant numeric">2</span>), <span class="constant numeric">1000</span>)[<span class="function call">OFFSET</span>(percentile <span class="keyword operator">*</span> <span class="constant numeric">10</span>)] <span class="keyword">AS</span> js_kbytes
<span class="keyword">FROM</span>
  <span class="string">`httparchive.summary_pages.2019_07_01_*`</span>,
  <span class="function call">UNNEST</span>([<span class="constant numeric">10</span>, <span class="constant numeric">25</span>, <span class="constant numeric">50</span>, <span class="constant numeric">75</span>, <span class="constant numeric">90</span>]) <span class="keyword">AS</span> percentile
<span class="keyword">GROUP</span> <span class="keyword">BY</span>
  percentile,
  client
<span class="keyword">ORDER</span> <span class="keyword">BY</span>
  percentile,
  client</code></pre>
```

**Review by bazzadp — CHANGES_REQUESTED** [review]
Thanks @chengxicn but spotted a small number of changes that need to be done.

**Comment by chengxicn** [comment]
Can't pass Test website after commit,   INFO:werkzeug:127.0.0.1 - - [07/Dec/2020 02:28:16] "GET /base/methodology HTTP/1.1" 308 -
DEBUG:root:Unsupported year requested: methodology

File: src/content/zh-CN/2020/capabilities.md [hunk 13]
```diff
@@ -0,0 +1,331 @@
...
+由于异步剪贴板API比较新，所以目前的使用率相当低。2020年3月，Safari在版本13.1中增加了对异步剪贴板API的支持。在2020年间，`read()`API的使用率越来越高。在2020年10月的谷歌Chrome浏览器中，0.0003%的页面加载过程中调用了该API。
+
+### 写访问
+
+除了读取操作外，异步剪贴板API还提供了两种向剪贴板写入内容的方法。同样，有一个针对纯文本的速记方法，叫做`navigator.clipboard.writeText()`，还有一个针对任意数据的方法，叫做`navigator.clipboard.write()`。在基于Chromium的浏览器中，当标签页处于活动状态时，向剪贴板写入内容不需要权限。但是，当网站处于后台时，试图向剪贴板写入数据则需要权限。由于这种方法需要先有用户的动作和权限，所以不在HTTP Archive指标的范围内。与`read()`方法相比，`write()`方法的使用量呈指数级增长，在2020年10月占了所有页面加载过程的0.0006%。
+
+[原始剪贴板访问API](https://bugs.chromium.org/p/chromium/issues/detail?id=897289)是Fugu的另一个能力，它甚至可以通过允许从剪贴板复制或粘贴任意数据来进一步增强异步剪贴板API。
+
+## 存储管理(<lang="en">StorageManager</lang>)API {存储管理storagemanagerapi}
```
Comment (bazzadp) [inline src/content/zh-CN/2020/capabilities.md:69]:
```suggestion
## 存储管理(<span lang="en">StorageManager</span>)API {存储管理storagemanagerapi}
```

File: src/content/zh-CN/2020/capabilities.md [hunk 14]
```diff
@@ -0,0 +1,329 @@
...
+  alt="使用 WebSocketStreams 的 Chrome 浏览器中的页面加载百分比",
+  caption='使用 WebSocketStreams 的 Chrome 浏览器中页面加载的百分比。<br>(来源: <a href="https://chromestatus.com/metrics/feature/timeline/popularity/3018">WebSocketStream</a>)',
+  description="WebSocketStreams 使用情况图，基于使用该功能的 Chrome 浏览器中页面加载的百分比。它显示了2020年6月和7月的峰值，其中API在大约0.0008%的页面加载中被使用。",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vTxqot9ALgxcgOVJntkzIKnkpo3idIPy-tL0t_nzC5BwFuq0ThgK5OXOYVVOpama4vB2EyggX813d33/pubchart?oid=1714443590&format=interactive",
+  sheets_gid="691106754"
+  )
+}}
+
+WebSocketStream API已经完成了第一次初期试用，现在又回到了实验阶段。这也解释了为什么目前这个API的使用率很低，几乎无法衡量。
+### 启用QUIC
```
Comment (bazzadp) [inline src/content/zh-CN/2020/capabilities.md:299]:
```suggestion

### 启用QUIC
```

File: src/content/zh-CN/2020/capabilities.md [hunk 15]
```diff
@@ -0,0 +1,331 @@
...
+```js
+self.addEventListener('periodicsync', (event) => {
+  if (event.tag === 'articles') {
+    event.waitUntil(syncStuff());
+  }
+});
+```
+
+在撰写本文时，只有基于Chromium的浏览器实现了这个API。在这些浏览器上，必须先安装应用程序（即添加到主屏幕上），然后才能使用该API。网站的[网站参与度评分](https://www.chromium.org/developers/design-documents/site-engagement)定义了是否可以调用周期性同步事件以及调用的频率。在目前保守的实现中，网站每天可以同步一次内容。
+{{ figure_markup(
```
Comment (bazzadp) [inline src/content/zh-CN/2020/capabilities.md:216]:
```suggestion

{{ figure_markup(
```

**Review by bazzadp — APPROVED** [review]

**Comment by bazzadp** [comment]
Thanks @chengxicn - excellent work as always!

**Pull request merged**